provider/generic_keycloak_identity_provider.go

```diff
@@ -0,0 +1,196 @@
+package provider
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/mrparkers/terraform-provider-keycloak/keycloak"
+	"strings"
+)
+
+var getIdentityProviderDataFunctions = map[string]func(data *schema.ResourceData) (*keycloak.IdentityProvider, error){
+	"oidc": getOidcIdentityProviderFromData,
+	"saml": getSamlIdentityProviderFromData,
+}
+
+var setIdentityProviderDataFunctions = map[string]func(data *schema.ResourceData, identityProvider *keycloak.IdentityProvider) error{
+	"oidc": setOidcIdentityProviderData,
+	"saml": setSamlIdentityProviderData,
+}
```

Comment:
I like the code re-use here, nice idea!

One suggestion - it seems a little strange that the file for the generic identity provider values has to be aware of the helper functions for the `oidc` and `saml` identity providers.  What do you think about passing those helper functions in as an argument to `resourceKeycloakIdentityProviderCreate`, etc instead?

I'll push a commit to your fork with my suggested changes.  If you agree, could you also refactor the mappers to follow this pattern as well?

Reply:
Right, I agree with that solution